.github/workflows/ci.yml

```diff
@@ -197,6 +197,119 @@ jobs:
     - name: Distcheck
       run: make distcheck DISTCHECK_CONFIGURE_FLAGS="--enable-werror"
 
+  build-dragonflybsd-latest-gcc:
+      runs-on: ubuntu-22.04
+      timeout-minutes: 20
+      steps:
+        - uses: actions/checkout@v4
+          with:
+            submodules: recursive
+        - name: Compile
+          uses: vmactions/dragonflybsd-vm@v1
+          with:
+            release: '6.4.0'
+            usesh: true
+            prepare: |
+              pkg install -y gmake autoconf-2.71 automake ncurses git
+              git config --global --add safe.directory /home/runner/work/htop/htop
+            run: |
+              set -e
+              export LDFLAGS="-L/usr/local/lib"
+              export CFLAGS="-I/usr/local/include"
+              ./autogen.sh
+              ./configure --enable-unicode
```

Comment:
I see other platforms in this .yml file have `--enable-werror` option specified in `./configure`. Is it a good idea to have `--emable-werror` here as well? (I'm not sure; if it could pop out a lot of compiler warnings then it's probably not a good idea.)

Reply:
We should aim to build without warnings for a "default" build. Skimming over the build logs it doesn't look like we have too many warnings.

Two options:
1. Go ahead without `-enable-werror` for now and enable it as a new PR
2. Enable it now and see what breaks.

If simply enabling it now works without causing compilation errors for any of the new platforms, I'd suggest to enable it now. Otherwise it's best to have it split.

Reply:
I have added `--enable-werror` for *BSD / Solaris, let us see what happens.

Reply:
There was a minor change needed for OpenBSD, I have pushed in that change to see if it works.

@BenBE: The change is a one liner, would you like me to split the PR? (enable `--enable-werror` for OpenBSD alone along with the fix).

Reply:
Updated the PR accordingly.